tests/test_utils.py

```diff
@@ -0,0 +1,128 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+"""Test cases for utils."""
+
+import copy
+import itertools
+import unittest
+
+import astropy.time
+import astropy.units as u
+import lsst.afw.image as afwImage
+import lsst.geom as geom
+import lsst.utils.tests
+import numpy as np
+from astro_metadata_translator import makeObservationInfo
+from lsst.obs.base import createInitialSkyWcsFromBoresight
+from lsst.obs.base.makeRawVisitInfoViaObsInfo import MakeRawVisitInfoViaObsInfo
+from lsst.obs.lsst.latiss import Latiss
+from lsst.summit.utils.utils import getExpPositionOffset
+
+
+class ExpSkyPositionOffsetTestCase(lsst.utils.tests.TestCase):
+    """A test case for testing sky position offsets for exposures."""
+
+    def setUp(self):
+        camera = Latiss.getCamera()
+        self.assertTrue(len(camera) == 1)
+        self.detector = camera[0]
+
+        self.viMaker = MakeRawVisitInfoViaObsInfo()
+        self.mi = afwImage.maskedImage.MaskedImageF(0, 0)
+        self.baseHeader = dict(boresight_airmass=1.5,
+                               temperature=15*u.deg_C,
+                               observation_type="science",
+                               exposure_time=5*u.ks,
+                               detector_num=32,
+                               )
+
+    def test_getExpPositionOffset(self):
+        epsilon = 0.0001
+        ra1s = [0, 45, 90]
+        ra2s = copy.copy(ra1s)
+        ra2s.extend([r + epsilon for r in ra1s])
+        ra1s = np.deg2rad(ra1s)
+        ra2s = np.deg2rad(ra2s)
+
+        epsilon = 0.0001
+        dec1s = [0, 45, 90]
+        dec2s = copy.copy(dec1s)
+        dec2s.extend([d + epsilon for d in dec1s[:-1]])  # skip last point as >90 not allowed for dec
+
+        rotAngle1 = geom.Angle(43.2, geom.degrees)  # arbitrary non-zero
+        rotAngle2 = geom.Angle(56.7, geom.degrees)
+
+        t1 = astropy.time.Time("2021-09-15T12:00:00", format="isot", scale="utc")
+        t2 = astropy.time.Time("2021-09-15T12:01:00", format="isot", scale="utc")
+        expTime = astropy.time.TimeDelta(20, format='sec')
+
+        header1 = copy.copy(self.baseHeader)
+        header2 = copy.copy(self.baseHeader)
+        header1['datetime_begin'] = astropy.time.Time(t1, format="isot", scale="utc")
+        header2['datetime_begin'] = astropy.time.Time(t2, format="isot", scale="utc")
+
+        header1['datetime_end'] = astropy.time.Time(t1+expTime, format="isot", scale="utc")
+        header2['datetime_end'] = astropy.time.Time(t2+expTime, format="isot", scale="utc")
+
+        obsInfo1 = makeObservationInfo(**header1)
+        obsInfo2 = makeObservationInfo(**header2)
+
+        vi1 = self.viMaker.observationInfo2visitInfo(obsInfo1)
+        vi2 = self.viMaker.observationInfo2visitInfo(obsInfo2)
```

Comment:
Use https://github.com/lsst/obs_base/blob/main/python/lsst/obs/base/makeRawVisitInfoViaObsInfo.py#L109

Reply:
Huh? I am, on that very line, no? `self.viMaker` _is_ `MakeRawVisitInfoViaObsInfo`

Reply:
but also there's no long any problems here.

Reply:
The `observationInfo2visitInfo` takes the ObservationInfo object you created above. The `__call__` method takes a header and first converts it to an ObservationInfo.

Reply:
I don't understand what you're asking me to do. I added `location=AUXTEL_LOCATION` to the base header and everything works.

Reply:
You are right. Sorry. Trying to comment whilst in a meeting.

Reply:
No probs. Merging 🎉